src/main/java/hillelJavaEE_02/doctor/DoctorRepository.java

```diff
@@ -0,0 +1,51 @@
+package hillelJavaEE_02.doctor;
+
+import hillelJavaEE_02.doctor.util.NoSuchSpecializationException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Repository;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ThreadLocalRandom;
+
+@Repository
+@RequiredArgsConstructor
+public class DoctorRepository {
+    private Map<Integer, Doctor> doctors = new ConcurrentHashMap<Integer, Doctor>() {{
+        put(0, new Doctor(0, "Doc", "surgeon"));
+        put(1, new Doctor(1, "Alex", "doctor"));
+    }};
+
+    private final DoctorConfig doctorConfig;

```

Comment:
I would say repository is wrong place for this entity. I would move it either to controller or to service.